pkg/config/schema/kubeclient/common.go

```diff
@@ -0,0 +1,152 @@
+	Informers() informerfactory.InformerFactory
+}
+
+func GetInformerFiltered[T runtime.Object](c ClientGetter, opts ktypes.InformerOptions) informerfactory.StartableInformer {
+	return GetInformerFilteredFromGVR(c, opts, kubetypes.GetGVR[T]())
+}
+
+func GetInformerFilteredFromGVR(c ClientGetter, opts ktypes.InformerOptions, g schema.GroupVersionResource) informerfactory.StartableInformer {
+	switch opts.InformerType {
+	case ktypes.DynamicInformer:
+		return getInformerFilteredDynamic(c, opts, g)
+	case ktypes.MetadataInformer:
+		return getInformerFilteredMetadata(c, opts, g)
+	default:
+		return getInformerFiltered(c, opts, g)
+	}
+}
+
+func getInformerFilteredDynamic(c ClientGetter, opts ktypes.InformerOptions, g schema.GroupVersionResource) informerfactory.StartableInformer {
+	return c.Informers().InformerFor(g, opts, func() cache.SharedIndexInformer {
+		inf := cache.NewSharedIndexInformerWithOptions(
+			&cache.ListWatch{
+				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+					options.FieldSelector = opts.FieldSelector
+					options.LabelSelector = opts.LabelSelector
+					return c.Dynamic().Resource(g).Namespace(opts.Namespace).List(context.Background(), options)
+				},
+				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+					return c.Dynamic().Resource(g).Namespace(opts.Namespace).Watch(context.Background(), options)
+				},
+			},
+			&unstructured.Unstructured{},
+			cache.SharedIndexInformerOptions{
+				ResyncPeriod:      0,
+				Indexers:          cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+				ObjectDescription: g.String(),
+			},
+		)
+		setupInformer(opts, inf)
+		return inf
+	})
+}
+
+func getInformerFilteredMetadata(c ClientGetter, opts ktypes.InformerOptions, g schema.GroupVersionResource) informerfactory.StartableInformer {
+	return c.Informers().InformerFor(g, opts, func() cache.SharedIndexInformer {
+		inf := cache.NewSharedIndexInformerWithOptions(
+			&cache.ListWatch{
+				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+					options.FieldSelector = opts.FieldSelector
+					options.LabelSelector = opts.LabelSelector
+					return c.Metadata().Resource(g).Namespace(opts.Namespace).List(context.Background(), options)
+				},
+				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+					return c.Metadata().Resource(g).Namespace(opts.Namespace).Watch(context.Background(), options)
+				},
+			},
+			&metav1.PartialObjectMetadata{},
+			cache.SharedIndexInformerOptions{
+				ResyncPeriod:      0,
+				Indexers:          cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+				ObjectDescription: g.String(),
+			},
+		)
+		setupInformer(opts, inf)
+		return inf
+	})
+}
+
+func setupInformer(opts ktypes.InformerOptions, inf cache.SharedIndexInformer) {
+	// It is important to set this in the newFunc rather than after InformerFor to avoid
+	// https://github.com/kubernetes/kubernetes/issues/117869
```

Comment:
There can be a caveat, if we want to create two informers with only different `ObjectTransform`. Actually the informer key does not consider it. So eventually we get only one informer

Reply:
I think this is best handled with logging. Its actually strictly worse to create 2 I think. But we should make sure we don't do this since we may miss optimizations, so I will make it a debug assertions so we can catch it in CI. I think we already have this in another part of the code, but I can make it cleaner now with this refactoring

Reply:
Fixed with last commit